packages/edit-widgets/src/hooks/use-last-selected-widget-area.js

```diff
@@ -21,17 +21,29 @@ const useLastSelectedRootId = () => {
 	}, [] );
 
 	const selectedRootId = useSelect( ( select ) => {
-		const { getBlockRootClientId, getBlockSelectionEnd } = select(
+		const { getBlockSelectionEnd, getBlockParents, getBlockName } = select(
 			'core/block-editor'
 		);
-		const blockSelectionEnd = getBlockSelectionEnd();
-		const blockRootClientId = getBlockRootClientId( blockSelectionEnd );
-		// getBlockRootClientId returns an empty string for top-level blocks, in which case just return the block id.
-		return blockRootClientId === '' ? blockSelectionEnd : blockRootClientId;
+		const blockSelectionEndClientId = getBlockSelectionEnd();
+
+		if (
+			getBlockName( blockSelectionEndClientId ) === 'core/widget-area'
+		) {
+			return blockSelectionEndClientId;
+		}
+
+		const blockParents = getBlockParents( blockSelectionEndClientId );
+		const rootWidgetAreaClientId = blockParents.find(
+			( clientId ) => getBlockName( clientId ) === 'core/widget-area'
+		);
+
+		if ( rootWidgetAreaClientId ) {
+			return rootWidgetAreaClientId;
+		}
```

Comment:
I made some changes here, as this is used with the `setIsWidgetAreaOpen` function, but previously could return a rootClientId for a block that's not a widget area (e.g. a group block).